src/optimizer/optimizer.cpp

```diff
@@ -113,6 +184,18 @@ std::unique_ptr<planner::AbstractPlanNode> Optimizer::ChooseBestPlan(
                                       std::move(children_plans), std::move(children_expr_map));
   OPTIMIZER_LOG_TRACE("Finish Choosing best plan for group " + std::to_string(id.UnderlyingValue()));
 
+  if (op->Contents()->GetOpType() == OpType::CTESCAN && !child_groups.empty()) {
+    NOISEPAGE_ASSERT(child_groups.size() <= 2, "CTE should not have more than 2 children.");
```

Comment:
what are valid numbers of children to have? 0? 1? 2? could you please document each case happens?